pandas/tests/resample/test_timedelta.py

```diff
@@ -114,14 +115,68 @@ def test_resample_timedelta_values():
     # check that timedelta dtype is preserved when NaT values are
     # introduced by the resampling
 
-    times = timedelta_range("1 day", "4 day", freq="4D")
+    times = timedelta_range("1 day", "6 day", freq="4D")
     df = DataFrame({"time": times}, index=times)
 
-    times2 = timedelta_range("1 day", "4 day", freq="2D")
+    times2 = timedelta_range("1 day", "6 day", freq="2D")
     exp = Series(times2, index=times2, name="time")
     exp.iloc[1] = pd.NaT
 
     res = df.resample("2D").first()["time"]
     tm.assert_series_equal(res, exp)
     res = df["time"].resample("2D").first()
     tm.assert_series_equal(res, exp)
+
+
+@pytest.mark.parametrize(
+    "start, end, freq, resample_freq",
+    [
+        ("8H", "21h59min50s", "10S", "3H"),  # GH 30353 example
+        ("3H", "22H", "1H", "5H"),
+        ("527D", "5006D", "3D", "10D"),
+        ("1D", "10D", "1D", "2D"),  # GH 13022 example
+        # tests that worked before GH 33498:
+        ("8H", "21h59min50s", "10S", "2H"),
+        ("0H", "21h59min50s", "10S", "3H"),
+        ("10D", "85D", "D", "2D"),
+    ],
+)
+def test_resample_timedelta_edge_case(start, end, freq, resample_freq):
+    # GH 33498
+    # check that the timedelta bins does not contains an extra bin
+    idx = pd.timedelta_range(start=start, end=end, freq=freq)
+    s = pd.Series(np.arange(len(idx)), index=idx)
+    result = s.resample(resample_freq).min()
+    expected_index = pd.timedelta_range(freq=resample_freq, start=start, end=end)
+    tm.assert_index_equal(result.index, expected_index)
+    assert not np.isnan(result[-1])
+
+
+@pytest.mark.parametrize(
+    "start, end, freq",
+    [
+        ("1D", "10D", "2D"),
+        ("2D", "30D", "3D"),
+        ("2s", "50s", "5s"),
+        # tests that worked before GH 33498:
+        ("4D", "16D", "3D"),
+        ("8D", "16D", "40s"),
+    ],
+)
+def test_timedelta_range_freq_divide_end(start, end, freq):
+    # GH 33498 only the cases where `(end % freq) == 0` used to fail
+
+    def mock_timedelta_range(start=None, end=None, **kwargs):
+        epoch = pd.Timestamp(0)
+        if start is not None:
+            start = epoch + pd.Timedelta(start)
+        if end is not None:
+            end = epoch + pd.Timedelta(end)
+        result = pd.date_range(start=start, end=end, **kwargs) - epoch
+        result.freq = freq
+        return result
+
+    res = pd.timedelta_range(start=start, end=end, freq=freq)
+    exp = mock_timedelta_range(start=start, end=end, freq=freq)
+
+    tm.assert_index_equal(res, exp)
```

Comment:
can you add an assertion that res.freq == exp.freq (that should be checked by assert_index_equal, but isnt yet)